service/controller/app/resource/pause/resource.go

```diff
@@ -0,0 +1,61 @@
+package pause
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/annotation"
+	"github.com/giantswarm/app/v3/pkg/key"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/v4/pkg/controller/context/reconciliationcanceledcontext"
+)
+
+const (
+	Name = "pause"
+)
+
+// Config represents the configuration used to create a new chartstatus resource.
+type Config struct {
+	Logger micrologger.Logger
+}
+
+// Resource implements the chartstatus resource.
+type Resource struct {
+	logger micrologger.Logger
+}
+
+func New(config Config) (*Resource, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	r := &Resource{
+		logger: config.Logger,
+	}
+
+	return r, nil
+}
+
+func (r *Resource) Name() string {
+	return Name
+}
+
+func (r *Resource) ensure(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToApp(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if len(cr.GetAnnotations()) == 0 {
+		return nil
+	}
+
+	v := cr.GetAnnotations()[annotation.AppOperatorPaused]
+	if v == "true" {
```

Comment:
if the annotation has a key named `annotation.AppOperatorPaused`, do we still need to see the value whether it is `true` or not?

Reply:
This is how it's actually done in cluster-api: https://github.com/kubernetes-sigs/cluster-api/blob/dadd7bd6233edb1e422e7af282e923abff52f830/util/annotations/helpers.go#L35 so let's do it this way.